resources/Microsoft.DotNet.Dsc/Microsoft.DotNet.Dsc.psm1

```diff
@@ -382,28 +405,22 @@ class DotNetToolPackage
                 # in this case, we misuse revision if beta,alpha, rc are present and grab the highest revision
                 $installedVersion = Get-Semver -Version $currentState.Version
                 $currentVersion = Get-Semver -Version $this.Version
-                if ($currentVersion -gt $installedVersion)
+                if ($currentVersion -gt $installedVersion -or $currentState -le $currentState)
```

Comment:
```suggestion
                if ($currentVersion -gt $installedVersion -or $currentState -le $currentState)
```

`$currentState -le $currentState`? Not sure what the intent here is but this comparison seems to cover greater than, less than, and equal.

Can you simplify to `$currentVersion -ne $installedVersion`?

Reply:
Great catch Ryan. It has been changed. Tests have successfully run on my side. Please give it a go in the pipelines :)

Would love to see it get to alpha release on the gallery!